src/System.IO.Pipelines/System/Threading/Scheduler.cs

```diff
@@ -5,8 +5,11 @@ namespace System.Threading
 {
     public abstract class Scheduler
     {
-        public static Scheduler TaskRun { get; } = new TaskRunScheduler();
```

Comment:
Why this change?

Reply:
Because you said so here https://github.com/dotnet/corefxlab/pull/2026#pullrequestreview-87068486

Reply:
Should have been more specific, why the syntax change.

Reply:
Storing it in a `Scheduler` field means the exact type gets lost for:
```csharp
Scheduler.ThreadPool.Schedule((() => {})
```
So it remains a virtual call. 

Storing it in an exact type field `ThreadPoolScheduler` means when `Scheduler.ThreadPool` inlines the Jit knows exactly what it is and can devitalize it to a direct call.

Won't directly help pipelines as the Transports store the configured values in a `Scheduler` field; but will help anyone calling it directly.

Reply:
Ben, what do you know about guns or shooting them at that? Perhaps shooting yourself in the foot is an exercise the application is required to perform albeit under controlled circumstance; in short we are already using Task what's left to shoot for? I would hope for something over thread like 🥇 

http://net7mma.codeplex.com/SourceControl/latest#Concepts/Classes/Threading/Threading.cs

And

http://net7mma.codeplex.com/SourceControl/latest#Concepts/Classes/Hardware.cs

Now that we finally have unsafe and call indirect... (thanks me)

http://net7mma.codeplex.com/SourceControl/latest#Concepts/Classes/CommonIntermediateLanguage.cs

http://net7mma.codeplex.com/SourceControl/latest#Concepts/Classes/Unsafe.cs

*sigh*...

..."Of the greatest King of Kings am I. If any other want to know how great I am or where I lie, let them outdo me in thy works." [In the poem Diodorus becomes "a traveller from an antique land...]

And whence we met the traveller(s) from such a car away and antique land, 
They spake and said—“Two vast and trunkless legs of stone 
Stand in the desert, one of brick and one of stone. . . . Near them, on the sand, 
Half sunk a shattered visage lies, whose frown, 
And wrinkled lip, and sneer of cold command, 
Tell that its sculptor well those passions read 
Which yet survive, stamped on these lifeless things, 
The hand that mocked them, and the heart that fed; 
And on the pedestal, these words appear: 

The flood lifted up the fish on to the cliff-bank;the whale became sad, where he swam on the shingle.Whale's bone.Magi.
--

Here fight Titus and the Jews.
Here the inhabitants flee from Jerusalem.

Doom.
Hostages.

Romulus and Remus, two brothers: a she-wolf fed them in Rome city, far from their native land.

Here the horse stands above the mound of woe,
It suffers tribulation; just as to her Erta appointed anxiety,
A grave of grief, in sorrow and anguish of heart.

Wood. Biter. Rush.


My name is not important, King of Kings I am, important. 
Look on my Works, ye Mighty, and despair! 
Nothing beside remains. Round the decay 
Of that colossal Wreck, boundless and bare 
Thy lone and level sands stretch far away... 

After the water recedes take sand and lament with it as I have, set thine creation in the sun and let is bask; In three days time whence my clay is hardednded  bring it forth so I may give it great scrutiny and critique.